infra/cifuzz/filestore/gitlab/__init__.py

```diff
@@ -0,0 +1,134 @@
+# Copyright 2021 Google LLC
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#      http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+"""Empty filestore implementation for platforms that haven't implemented it."""
+import logging
+
+import shutil
+import os
+import json
+import tempfile
+
+import http_utils
+import filestore
+
+# pylint:disable=no-self-use,unused-argument
+
+
+class GitlabFilestore(filestore.BaseFilestore):
+  """Implementation of BaseFilestore using Gitlab job artifacts. Relies on
+  using a cache to get latest job id oto get a type of artifacts.
+  having a PRIVATE-TOKEN supplied to access the Gitlab API as seems
+  the only way to download an artifact from another job"""
+
+  def __init__(self, config):
+    super().__init__(config)
+    self.downloaded = set()
+    self.artifacts_dir = self.config.platform_conf.artifacts_dir
+    self.cache_dir = self.config.platform_conf.cache_dir
+    self.download_dir = self.config.platform_conf.download_dir
+    self.api_url = self.config.platform_conf.api_url
+
+  def _copy_from_dir(self, src, name, reason):
+    dest_dir = os.path.join(self.config.workspace, self.artifacts_dir, reason,
+                            name)
+    logging.info('Uploading %s to artifacts to %s.', reason, dest_dir)
+    shutil.copytree(src, dest_dir)
+    # Saves current job id in gitlab cache.
+    job_id = self.config.platform_conf.current_job_id
+    cache_file_path = os.path.join(self.config.workspace, self.cache_dir,
```

Comment:
Why does there need to be a different cache_dir for each reason. Isn't the cache just for storing the job dir?

Reply:
The cache just stores the job id, not the job dir.
But is stores multiple job ids for the multiple reasons.
A job producing the coverage report, will not produce a corpus